### PR TITLE
[ROCm] ROCm compatible configs for triton kernels

### PR DESCRIPTION
**Pull request by jataylo**
This PR brings in a few inductor changes required for ROCm

~**1 - Introduction of a toggle for enforced channel last convolution fallbacks**~
This addition is split off into its own PR after some cleanup by @pragupta  https://github.com/pytorch/pytorch/pull/107812

**2 - Addition of ROCm specific block sizes**
We are now able to support the MAX_AUTOTUNE mode on ROCm, we are proposing conditions to allow us to finetune our own block tuning. Currently triton on ROCm does not benefit from pipelining so we are setting all configs to `num_stages=1` and we have removed some upstream tunings on ROCm to avoid running out of shared memory resources.

In the future we will provide more optimised tunings for ROCm but for now this should mitigate any issues

~**3 - Addition of device_type to triton's compile_meta**~
~Proposing this addition to `triton_heuristics.py`, Triton on ROCm requires device_type to be set to hip https://github.com/ROCmSoftwarePlatform/triton/pull/284 suggesting to bring this change in here so we can pass down the correct device type to triton.~
This change is split off and will arrive in the wheel update PR https://github.com/pytorch/pytorch/pull/107600 leaving this PR to focus on the ROCm specific block sizes.

cc @jeffdaily @sunway513 @jithunnair-amd @pruthvistony @ROCmSupport @dllehr-amd @hongxiayang @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov

### CODE REVIEW

File: torch/_inductor/config.py
Comment (jithunnair-amd) [inline torch/_inductor/config.py:177]: @jataylo I think [PR 107477](https://github.com/pytorch/pytorch/pull/107477) should provide a more robust way of coding such GPU-type-conditional checks since I'm not sure if the `device_properties.name` field will always contain a reliable string to match on.

File: torch/_inductor/kernel/conv.py
Comment (jithunnair-amd) [inline torch/_inductor/kernel/conv.py:48]: @jataylo Please add a comment explaining why we are choosing different configs for ROCm.

File: torch/_inductor/kernel/mm_common.py
Comment (jithunnair-amd) [inline torch/_inductor/kernel/mm_common.py:47]: @jataylo Please add a comment explaining why we are choosing different configs for ROCm.

File: torch/_inductor/kernel/mm_plus_mm.py
Comment (jithunnair-amd) [inline torch/_inductor/kernel/mm_plus_mm.py:107]: @jataylo Please add a comment explaining why we are choosing different configs for ROCm.

**Review by jithunnair-amd — APPROVED** [review]
Please add comments

**Review by jithunnair-amd — CHANGES_REQUESTED** [review]
Actually, changing to "Request changes" until the conversation regarding PR 107477 as a dependency is resolved.

File: torch/_inductor/config.py
Comment (jataylo) [reply torch/_inductor/config.py:177]: Planning to replace with gcnArchName when that PR lands

**Comment by jataylo** [comment]
@jansel @eellison @ngimel 

Hi all opening this PR up for review. These are some additions we are hoping to get in before the 2.1 cutoff. Please let me know if you know of anywhere else that the nhwc toggle would need to cover.

File: torch/_inductor/kernel/conv.py
Comment (jithunnair-amd) [reply torch/_inductor/kernel/conv.py:48]: https://github.com/pytorch/pytorch/pull/107584/files#diff-3385fcbafbb6165d9e55060e0ddf92c46e17625eb6c0c38a07f32995950882daR63

File: torch/_inductor/kernel/mm_common.py
Comment (jithunnair-amd) [reply torch/_inductor/kernel/mm_common.py:47]: https://github.com/pytorch/pytorch/pull/107584/files#diff-8973a6537701d24baabc55ae4c64093f4762a30e4305fddbbbba9d4b54fca3a1R88

File: torch/_inductor/kernel/mm_plus_mm.py
Comment (jithunnair-amd) [reply torch/_inductor/kernel/mm_plus_mm.py:107]: https://github.com/pytorch/pytorch/pull/107584/files#diff-d7d9c568a332fc7a57858d6dfb60707dc04366bc5a3d2659c85836104f3aa9abR147

File: torch/_inductor/config.py
Comment (jithunnair-amd) [reply torch/_inductor/config.py:178]: Actually, to delink the PRs, let's go ahead with above logic slightly modified to match "MI2\d\d"

**Review by jithunnair-amd — APPROVED** [review]

File: torch/_inductor/kernel/mm_plus_mm.py
Comment (kit1980) [inline torch/_inductor/kernel/mm_plus_mm.py:147]: typo: Seperate

Comment (jithunnair-amd) [reply torch/_inductor/kernel/mm_plus_mm.py]: fixed

File: torch/_inductor/graph.py
Comment (eellison) [inline torch/_inductor/graph.py:374]: We should be making channels-last occur here.


https://github.com/pytorch/pytorch/blob/0522fa5f9676dfe5eabadc78f83b8876145c5bf4/torch/_inductor/graph.py#L248-L257

File: torch/_inductor/graph.py
Comment (eellison) [inline torch/_inductor/graph.py:676]: we still want the same debug logic to be printed even if the newly added config option is False

**Comment by atalman** [comment]
@jithunnair-amd there are couple of rocm failures
[linux-focal-rocm5.6-py3.8 / test (distributed, 1, 2, linux.rocm.gpu)](https://hud.pytorch.org/pr/pytorch/pytorch/107584#16122778366) ([gh](https://github.com/pytorch/pytorch/actions/runs/5944674251/job/16122778366))
[linux-focal-rocm5.6-py3.8 / test (distributed, 2, 2, linux.rocm.gpu)](https://hud.pytorch.org/pr/pytorch/pytorch/107584#16122778460) ([gh](https://github.com/pytorch/pytorch/actions/runs/5944674251/job/16122778460))
Can you please check ?

File: torch/_inductor/kernel/conv.py
Comment (eellison) [inline torch/_inductor/kernel/conv.py:279]: If the config to require channels last is working correctly, I dont think this is necessary.

File: torch/_inductor/kernel/conv.py
Comment (eellison) [inline torch/_inductor/kernel/conv.py:379]: this is making it so that we are no longer forcing channels last when `inductor_config.conv_force_channels_last` is False but `layout_opt` is true.

Also we should remove the inline logic for `conv_force_channels_last` and just take it into account in `decide_layout_opt`

File: torch/_inductor/config.py
Comment (jansel) [inline torch/_inductor/config.py:170]: I'd suggest moving this logic outside of `config.py` and into a helper function where it is used.  Perhaps allow 3 values: True/False/None where None defaults to the heuristic below.

File: torch/_inductor/kernel/mm_common.py
Comment (eellison) [inline torch/_inductor/kernel/mm_common.py:52]: Same question here, are we just setting num_stages=1 ?

**Review by jansel — CHANGES_REQUESTED** [review]

File: torch/_inductor/kernel/mm_plus_mm.py
Comment (eellison) [inline torch/_inductor/kernel/mm_plus_mm.py:148]: So we're skipping this config `{"BLOCK_M": 32, "BLOCK_N": 32, "BLOCK_K": 128},` and setting num_stages=1. Would it be worthwhile to post-process the configs instead of duplicating them ?

If there's a standard way of converting NVIDIA->AMD configs it'd be nice to have a function to do it.

**Review by eellison — COMMENTED** [review]
Looking good but we need to update how the channels last logic is done. We should only have to change the `decide_layout_opt` function instead of all the other code paths that are changed currently. 

Also had a question about the duplicate configs

File: torch/_inductor/kernel/mm_plus_mm.py
Comment (jataylo) [reply torch/_inductor/kernel/mm_plus_mm.py:148]: @eellison Thank you for the review, I agree the config duplication is an issue, especially if some other conditional logic arrives later on

At the moment I cannot think of a nice way to have a function to convert NVIDIA to AMD configs, although the num_warps=1 change is trivial we would require a good approximation for triton's shared memory usage for each kernel to filter out configs that could break on our side. In the future we would also like the option of providing our own relevant tunings that will only run on ROCm.

I can think of two atlernatives that may be cleaner

1. We create our own `rocm_configs` file and we can simply pull the configs from here instead of listing in the file.

2. We could extend the list of arrays to incorporate a condition for each config then perform a list comprehension to get a filtered configs list. 

For example with a list of dictionaries
```
# Dictionary to store conv kernel configs - "cond": True means that
# the config is used by default but any arbitrary condition can be applied
# here for a specific platform or use case
all_configs = [
    # "BLOCK_M", "BLOCK_N", "BLOCK_K", "NUM_STAGES", "NUM_WARPS"
    {"config": (64, 256, 16, 2, 4), "cond": True}, 
    {"config": (16, 64, 16, 1, 4), "cond": True},
    {"config": (1024, 16, 16, 1, 8), "cond": True},
    {"config": (128, 128, 32, 2, 8), "cond": True},
    {"config": (64, 64, 32, 2, 4), "cond": True}, 
    {"config": (64, 256, 32, 2, 8), "cond": True}, 
    {"config": (256, 64, 32, 2, 8), "cond": True},
    {"config": (16, 16, 16, 1, 8), "cond": torch.version.hip},
]

# Create filtered list of configs from conv_configs based on cond
configs = [config["config"] for config in all_configs if config["cond"]]

conv_configs = functools.partial(
    filtered_configs,
    configs=configs,
)
```

This would at least let us be able to maintain a single list and support conditions on platforms but could end up being a bulky structure if many configs are added

File: torch/_inductor/config.py
Comment (jataylo) [reply torch/_inductor/config.py]: Thank you for the review @jansel

@pragupta was able to take the comment and submit a PR which is much cleaner using layout_opt could you take a look at https://github.com/pytorch/pytorch/pull/107812

This PR will now focus on adding the device_type into compile_meta and getting our ROCm compatible triton configs

File: torch/_inductor/kernel/mm_plus_mm.py
Comment (jataylo) [reply torch/_inductor/kernel/mm_plus_mm.py]: Hey @eellison I have updated this PR which hopefully will alleviate the concerns on duplicated configs, please could you take a look at the latest update? 

I've changed the base config format to be a list of dictionaries which will allow us provide conditional logic for individual configs for fine tuning on arbitrary platforms

I ran some sanity checks locally to ensure everything was as expected. 
```
**conv_configs**
(BEFORE CHANGE)
(
(64, 256, 16, 2, 4), 
(256, 64, 16, 2, 4), 
(1024, 16, 16, 1, 8), 
(128, 128, 32, 2, 8), 
(64, 64, 32, 2, 4), 
(64, 256, 32, 2, 8), 
(256, 64, 32, 2, 8))

(AFTER CHANGE - None rocm)
(
(64, 256, 16, 2, 4), 
(256, 64, 16, 2, 4), 
(1024, 16, 16, 1, 8), 
(128, 128, 32, 2, 8), 
(64, 64, 32, 2, 4), 
(64, 256, 32, 2, 8), 
(256, 64, 32, 2, 8)
)

(AFTER CHANGE - ROCm)
(
(64, 256, 16, 1, 4), 
(256, 64, 16, 1, 4), 
(1024, 16, 16, 1, 8), 
(128, 128, 32, 1, 8), 
(64, 64, 32, 1, 4), 
(64, 256, 32, 1, 8), 
(256, 64, 32, 1, 8)
)



mm_configs

(BEFORE CHANGE)
mm_configs
((64, 64, 32, 2, 4), 
(64, 128, 32, 3, 4), 
(128, 64, 32, 3, 4), 
(64, 128, 32, 4, 8), 
(128, 64, 32, 4, 8), 
(64, 32, 32, 5, 8), 
(32, 64, 32, 5, 8), 
(128, 128, 32, 2, 8), 
(64, 64, 64, 3, 8), 
(32, 32, 128, 2, 4), 
(64, 64, 16, 2, 4), 
(32, 32, 16, 1, 2))

int8_mm_configs
(
(64, 64, 32, 2, 4), 
(64, 128, 32, 3, 4), 
(128, 64, 32, 3, 4), 
(64, 128, 32, 4, 8), 
(128, 64, 32, 4, 8), 
(64, 32, 32, 5, 8), 
(32, 64, 32, 5, 8), 
(128, 128, 32, 2, 8), 
(64, 64, 64, 3, 8), 
(128, 256, 128, 3, 8), 
(256, 128, 128, 3, 8)
)

(AFTER CHANGE - No ROCm)
mm_configs
(
(64, 64, 32, 2, 4), 
(64, 128, 32, 3, 4), 
(128, 64, 32, 3, 4), 
(64, 128, 32, 4, 8), 
(128, 64, 32, 4, 8), 
(64, 32, 32, 5, 8), 
(32, 64, 32, 5, 8), 
(128, 128, 32, 2, 8), 
(64, 64, 64, 3, 8), 
(32, 32, 128, 2, 4), 
(64, 64, 16, 2, 4), 
(32, 32, 16, 1, 2)
)

int8_configs
(
(64, 64, 32, 2, 4), 
(64, 128, 32, 3, 4), 
(128, 64, 32, 3, 4), 
(64, 128, 32, 4, 8), 
(128, 64, 32, 4, 8), 
(64, 32, 32, 5, 8), 
(32, 64, 32, 5, 8), 
(128, 128, 32, 2, 8), 
(64, 64, 64, 3, 8), 
(128, 256, 128, 3, 8), 
(256, 128, 128, 3, 8)
)


(AFTER CHANGE - ROCm)
mm_configs
(
(64, 64, 32, 1, 4), 
(64, 128, 32, 1, 4), 
(128, 64, 32, 1, 4), 
(64, 128, 32, 1, 8), 
(128, 64, 32, 1, 8), 
(64, 32, 32, 1, 8), 
(32, 64, 32, 1, 8), 
(128, 128, 32, 1, 8), 
(64, 64, 64, 1, 8), 
(64, 64, 16, 1, 4), 
(32, 32, 16, 1, 2)
)

(AFTER CHANGE - ROCM)
int8_mm_configs
(
(64, 64, 32, 1, 4), 
(64, 128, 32, 1, 4), 
(128, 64, 32, 1, 4), 
(64, 128, 32, 1, 8), 
(128, 64, 32, 1, 8), 
(64, 32, 32, 1, 8), 
(32, 64, 32, 1, 8), 
(128, 128, 32, 1, 8), 
(64, 64, 64, 1, 8)
)

mm_plus_mm

Original:
[<triton.runtime.autotuner.Config object at 0x7f9dcfc4e1d0>, <triton.runtime.autotuner.Config object at 0x7f9dcfc4e5f0>, <triton.runtime.autotuner.Config object at 0x7f9dcfc4d270>, <triton.runtime.autotuner.Config object at 0x7f9dcfb579d0>, <triton.runtime.autotuner.Config object at 0x7f9dcfb57790>, <triton.runtime.autotuner.Config object at 0x7f9dcfb57670>, <triton.runtime.autotuner.Config object at 0x7f9dcfb57970>, <triton.runtime.autotuner.Config object at 0x7f9dcfb57ac0>, <triton.runtime.autotuner.Config object at 0x7f9dcfb579a0>, <triton.runtime.autotuner.Config object at 0x7f9dcfb57a60>]

After change - no ROCm
[<triton.runtime.autotuner.Config object at 0x7f9dcfb57b50>, <triton.runtime.autotuner.Config object at 0x7f9dcfb57b80>, <triton.runtime.autotuner.Config object at 0x7f9dcfb57be0>, <triton.runtime.autotuner.Config object at 0x7f9dcfb57c40>, <triton.runtime.autotuner.Config object at 0x7f9dcfb57ca0>, <triton.runtime.autotuner.Config object at 0x7f9dcfb57d00>, <triton.runtime.autotuner.Config object at 0x7f9dcfb57d60>, <triton.runtime.autotuner.Config object at 0x7f9dcfb57dc0>, <triton.runtime.autotuner.Config object at 0x7f9dcfb57e20>, <triton.runtime.autotuner.Config object at 0x7f9dcfb57e80>]

MAX_AUTOTUNE log to verify
ExternKernelCaller(extern_kernels._mm_plus_mm)
TritonTemplateCaller(/tmp/torchinductor_root/w7/cw7tibckrubr4blgzo2wxqsuszcu6kktg3m3ulbywdjdx3loekyf.py, ACC_TYPE='tl.float32', ALLOW_TF32=False, BLOCK_K=32, BLOCK_M=64, BLOCK_N=64, B_PROLOGUE_CAST_TYPE=None, EVEN_K=True, GROUP_M=8, num_stages=2, num_warps=4)
TritonTemplateCaller(/tmp/torchinductor_root/zi/cziadg6jiurq6wi5c3sml3d2bngkgyldg3bvzekwmzgdynvqccjf.py, ACC_TYPE='tl.float32', ALLOW_TF32=False, BLOCK_K=32, BLOCK_M=64, BLOCK_N=64, B_PROLOGUE_CAST_TYPE=None, EVEN_K=True, GROUP_M=8, num_stages=3, num_warps=8)
TritonTemplateCaller(/tmp/torchinductor_root/jv/cjvicdxe77ywsfgipbipnobl7sg25fnd4luplmcxkg53jxyfsji5.py, ACC_TYPE='tl.float32', ALLOW_TF32=False, BLOCK_K=32, BLOCK_M=64, BLOCK_N=64, B_PROLOGUE_CAST_TYPE=None, EVEN_K=True, GROUP_M=8, num_stages=4, num_warps=16)
TritonTemplateCaller(/tmp/torchinductor_root/oo/cooaupuodzdjzlzsq6hk6p3q6sbfwxjxjj3il4yxot7fqeq4zyet.py, ACC_TYPE='tl.float32', ALLOW_TF32=False, BLOCK_K=32, BLOCK_M=64, BLOCK_N=32, B_PROLOGUE_CAST_TYPE=None, EVEN_K=True, GROUP_M=8, num_stages=4, num_warps=8)
TritonTemplateCaller(/tmp/torchinductor_root/u7/cu7wncwk4rgrmz5ibsywzozc73a4x4xgg46cetvu7xohakeduaxu.py, ACC_TYPE='tl.float32', ALLOW_TF32=False, BLOCK_K=32, BLOCK_M=32, BLOCK_N=64, B_PROLOGUE_CAST_TYPE=None, EVEN_K=True, GROUP_M=8, num_stages=4, num_warps=8)
TritonTemplateCaller(/tmp/torchinductor_root/3p/c3pmqu4wxiri3pctkeq7ugzjmta75xg522wka76akkgnqlkhtvjw.py, ACC_TYPE='tl.float32', ALLOW_TF32=False, BLOCK_K=32, BLOCK_M=128, BLOCK_N=128, B_PROLOGUE_CAST_TYPE=None, EVEN_K=True, GROUP_M=8, num_stages=1, num_warps=8)
TritonTemplateCaller(/tmp/torchinductor_root/ff/cffv5gxhrk5mll53j2w4hvoeclaumnjruxonmto4o34dn2zgqmn6.py, ACC_TYPE='tl.float32', ALLOW_TF32=False, BLOCK_K=64, BLOCK_M=64, BLOCK_N=64, B_PROLOGUE_CAST_TYPE=None, EVEN_K=True, GROUP_M=8, num_stages=1, num_warps=8)
TritonTemplateCaller(/tmp/torchinductor_root/bs/cbs7rekdixkkbypx7eomq3s7yntk7dqqt6klbcelxyrbntn4gvmd.py, ACC_TYPE='tl.float32', ALLOW_TF32=False, BLOCK_K=128, BLOCK_M=32, BLOCK_N=32, B_PROLOGUE_CAST_TYPE=None, EVEN_K=True, GROUP_M=8, num_stages=1, num_warps=8)
TritonTemplateCaller(/tmp/torchinductor_root/wn/cwnc74qawqd2fpt45sqbsm2hzijhtoek6cr7o75apy5ks2eslxej.py, ACC_TYPE='tl.float32', ALLOW_TF32=False, BLOCK_K=16, BLOCK_M=64, BLOCK_N=64, B_PROLOGUE_CAST_TYPE=None, EVEN_K=True, GROUP_M=8, num_stages=2, num_warps=4)
TritonTemplateCaller(/tmp/torchinductor_root/nc/cncaszarsw2mxvumukoqtoqmedmfj6dyccev6pgyzbztbxyah5ng.py, ACC_TYPE='tl.float32', ALLOW_TF32=False, BLOCK_K=16, BLOCK_M=32, BLOCK_N=32, B_PROLOGUE_CAST_TYPE=None, EVEN_K=True, GROUP_M=8, num_stages=1, num_warps=2)



After change - ROCm
[<triton.runtime.autotuner.Config object at 0x7f3ad952fe50>, <triton.runtime.autotuner.Config object at 0x7f3ad952feb0>, <triton.runtime.autotuner.Config object at 0x7f3ad952ff10>, <triton.runtime.autotuner.Config object at 0x7f3ad952ff70>, <triton.runtime.autotuner.Config object at 0x7f3ad952ffd0>, <triton.runtime.autotuner.Config object at 0x7f3ad9578070>, <triton.runtime.autotuner.Config object at 0x7f3ad95780d0>, <triton.runtime.autotuner.Config object at 0x7f3ad9578130>, <triton.runtime.autotuner.Config object at 0x7f3ad9578190>]

MAX_AUTOTUNE log to verify:
ExternKernelCaller(extern_kernels._mm_plus_mm)
TritonTemplateCaller(/tmp/torchinductor_root/e3/ce3giwk6mjj4doboazushhkrlbuqg2bdsxp64fpr4bszl4vrt2tb.py, ACC_TYPE='tl.float32', ALLOW_TF32=False, BLOCK_K=32, BLOCK_M=64, BLOCK_N=64, B_PROLOGUE_CAST_TYPE=None, EVEN_K=True, GROUP_M=8, num_stages=1, num_warps=4)
TritonTemplateCaller(/tmp/torchinductor_root/hl/chlf74iccwcztxrqyeclfytc4y5iglfkap43kyuuarshh4zp3zjq.py, ACC_TYPE='tl.float32', ALLOW_TF32=False, BLOCK_K=32, BLOCK_M=64, BLOCK_N=64, B_PROLOGUE_CAST_TYPE=None, EVEN_K=True, GROUP_M=8, num_stages=1, num_warps=8)
TritonTemplateCaller(/tmp/torchinductor_root/si/csie65r5karqtew34dxaoevjjbe3nr4zbg7sxw6tzoz5o7wmlk6d.py, ACC_TYPE='tl.float32', ALLOW_TF32=False, BLOCK_K=32, BLOCK_M=64, BLOCK_N=64, B_PROLOGUE_CAST_TYPE=None, EVEN_K=True, GROUP_M=8, num_stages=1, num_warps=16)
TritonTemplateCaller(/tmp/torchinductor_root/r7/cr7gmanqx5yda5fpjq7mqu7gipgmbsx4tth32v522oxh4q6kkodk.py, ACC_TYPE='tl.float32', ALLOW_TF32=False, BLOCK_K=32, BLOCK_M=64, BLOCK_N=32, B_PROLOGUE_CAST_TYPE=None, EVEN_K=True, GROUP_M=8, num_stages=1, num_warps=8)
TritonTemplateCaller(/tmp/torchinductor_root/yl/cylqmexqqa4qn72tyqdg4lqxvvb6pnelked5d6tndk67daxxn4ps.py, ACC_TYPE='tl.float32', ALLOW_TF32=False, BLOCK_K=32, BLOCK_M=32, BLOCK_N=64, B_PROLOGUE_CAST_TYPE=None, EVEN_K=True, GROUP_M=8, num_stages=1, num_warps=8)
TritonTemplateCaller(/tmp/torchinductor_root/3p/c3pmqu4wxiri3pctkeq7ugzjmta75xg522wka76akkgnqlkhtvjw.py, ACC_TYPE='tl.float32', ALLOW_TF32=False, BLOCK_K=32, BLOCK_M=128, BLOCK_N=128, B_PROLOGUE_CAST_TYPE=None, EVEN_K=True, GROUP_M=8, num_stages=1, num_warps=8)
TritonTemplateCaller(/tmp/torchinductor_root/ff/cffv5gxhrk5mll53j2w4hvoeclaumnjruxonmto4o34dn2zgqmn6.py, ACC_TYPE='tl.float32', ALLOW_TF32=False, BLOCK_K=64, BLOCK_M=64, BLOCK_N=64, B_PROLOGUE_CAST_TYPE=None, EVEN_K=True, GROUP_M=8, num_stages=1, num_warps=8)
TritonTemplateCaller(/tmp/torchinductor_root/uw/cuwh572otqokk5n4fbjaaci3bbhb5mhfoiklphkazueslrbm6np4.py, ACC_TYPE='tl.float32', ALLOW_TF32=False, BLOCK_K=16, BLOCK_M=64, BLOCK_N=64, B_PROLOGUE_CAST_TYPE=None, EVEN_K=True, GROUP_M=8, num_stages=1, num_warps=4)
TritonTemplateCaller(/tmp/torchinductor_root/nc/cncaszarsw2mxvumukoqtoqmedmfj6dyccev6pgyzbztbxyah5ng.py, ACC_TYPE='tl.float32', ALLOW_TF32=False, BLOCK_K=16, BLOCK_M=32, BLOCK_N=32, B_PROLOGUE_CAST_TYPE=None, EVEN_K=True, GROUP_M=8, num_stages=1, num_warps=2)
```

File: torch/_inductor/kernel/mm_common.py
Comment (jataylo) [reply torch/_inductor/kernel/mm_common.py]: Updated to avoid duplication and allow for conditional logic (see comment below for more information)

File: torch/_inductor/kernel/conv.py
Comment (jataylo) [reply torch/_inductor/kernel/conv.py]: @pragupta was able to take the comment and submit a PR which is much cleaner using layout_opt could you take a look at https://github.com/pytorch/pytorch/pull/107812

This PR will now focus on adding the device_type into compile_meta and getting our ROCm compatible triton configs

File: torch/_inductor/kernel/conv.py
Comment (jataylo) [reply torch/_inductor/kernel/conv.py]: @pragupta was able to take the comment and submit a PR which is much cleaner using layout_opt could you take a look at https://github.com/pytorch/pytorch/pull/107812

This PR will now focus on adding the device_type into compile_meta and getting our ROCm compatible triton configs

File: torch/_inductor/graph.py
Comment (jataylo) [reply torch/_inductor/graph.py]: @pragupta was able to take the comment and submit a PR which is much cleaner using layout_opt could you take a look at https://github.com/pytorch/pytorch/pull/107812

This PR will now focus on adding the device_type into compile_meta and getting our ROCm compatible triton configs

I've also submitted an update to allow for conditional logic in the conv/mm configs which should avoid duplication, could get your review on this?

**Review by jansel — APPROVED** [review]

**Review by eellison — APPROVED** [review]

**Comment by jataylo** [comment]
Fixing merge conflict, some ROCm tests didn't kick off last run, leaving for CI once again. Once green will merge.

**Comment by jataylo** [comment]
Trying to retrigger this distributed rocm shard that isn't running UTs. Will merge after.

**Comment by huydhn** [comment]
@pytorchbot drci

(Please ignore this comment, I'm testing Dr.CI)

**Pull request closed without merging**
